pkg/docker/config/config_test.go

```diff
@@ -58,13 +58,13 @@ func TestGetPathToAuth(t *testing.T) {
 			if c.xrd == "" {
 				os.Unsetenv("XDG_RUNTIME_DIR")
 			}
-			res, lf, err := getPathToAuthWithOS(c.sys, c.os)
+			res, err := getPathToAuthWithOS(c.sys, c.os)
 			if c.expected == "" {
 				assert.Error(t, err)
 			} else {
 				require.NoError(t, err)
-				assert.Equal(t, c.expected, res)
-				assert.Equal(t, c.legacyFormat, lf)
+				assert.Equal(t, c.expected, res.path)
+				assert.Equal(t, c.legacyFormat, res.legacyFormat)
```

Comment:
Absolutely non-blocking:
```go
assert.Equal(t, authPath{path: c.expected, legacyFormat: c.legacyFormat}, res)
```
would work, and going further `c.expected` could be a whole `authPath` value instead of individual fields.

This works just fine as is.